src/ShaderManager.cpp

```diff
@@ -178,7 +178,9 @@ void ShaderManager::renderObject(std::shared_ptr<GameObject> objToRender, const
 
 		M->translate(objToRender->getPosition());
 		M->scale(objToRender->getScale());
-		M->rotate(objToRender->getYAxisRotation(), glm::vec3(0.0f, 1.0f, 0.0f));
+
+      glm::mat4 rotation = objToRender->transform.getRotate();
+		M->rotateMat4(rotation);
 
```

Comment:
The reason for the extra abstraction here was just to remove any need to know how the object was maintaining it's transforms. However, since it looks like we're going full boar on MatrixTransform vs MatrixStack (which is fine) I'm okay with simplifying everything to just use the transform I think.